### ISSUE TITLE
View.toBitmap should take scroll state into consideration

### ISSUE THREAD

**Issue by LanderlYoung** [android/android-ktx#500]
The following code (in View.kt) does not take what's on the screen into the bitmap, especially when the view is a ViewPager or ScrollView.

```kotlin
fun View.toBitmap(config: Bitmap.Config = Bitmap.Config.ARGB_8888): Bitmap {
    if (!ViewCompat.isLaidOut(this)) {
        throw IllegalStateException("View needs to be laid out before calling toBitmap()")
    }
    return Bitmap.createBitmap(width, height, config).applyCanvas(::draw)
}
```

Instead, it's better to be something like this:

```kotlin
fun View.toBitmap(config: Bitmap.Config = Bitmap.Config.ARGB_8888): Bitmap {
    if (!ViewCompat.isLaidOut(this)) {
        throw IllegalStateException("View needs to be laid out before calling toBitmap()")
    }
    return Bitmap.createBitmap(width, height, config).applyCanvas { 
                    it.translate(-scrollX, -scrollY)
                    draw(it)
              }
}
```

**Comment by LanderlYoung** [comment]
Great! PR will be coming very soon!

**Comment by LanderlYoung** [comment]
PR submitted https://github.com/android/android-ktx/pull/504 !

**Comment by romainguy** [comment]
Fixed in PR #504
